src/server/models/helpers.ts

```diff
@@ -23,3 +25,20 @@ export const rawInsertGeoLocation = async (
 export function geoPointIsValid(geoPoint: any): boolean {
   return isArray(geoPoint) && isNumber(geoPoint[0]) && isNumber(geoPoint[1]);
 }
+
+export async function getListIdForCountryAndType(
+  country: string,
+  serviceType: ServiceType
+): Promise<number> {
+  const existingLists = await findListByCountryAndType(
+    country,
+    serviceType
+  );
+
+  if (existingLists !== undefined && existingLists?.length > 0) {
```

Comment:
you can just do `existingLists?.length` without doing an undefined check. If existingLists is undefined or null, it will stop executing and return undefined (which is falsy). Actually, judging by this if block, you can remove the if statement entirely

Reply:
for the if statement you state we should be able to remove it entirely.  Doing this could produce an indexOutOfBoundsError (or equivalent) if the array is empty (via `const list = existingLists[0];`) .  We also need a number (default -1) to be populated in the ListItem.listId column which is why the `return -1` is included at the end.  While there should always be a match, as a guard condition I think we should keep it there.